pkg/basicstation/cups/messages.go

```diff
@@ -84,6 +90,17 @@ func TokenCredentials(trust *x509.Certificate, authorization string) ([]byte, er
 	out = append(out, trust.Raw...)
 	// TODO: Refactor when client side TLS is supported https://github.com/TheThingsNetwork/lorawan-stack/issues/137
 	out = append(out, emptyClientCert...)
+	token := authorization
+	if strings.HasPrefix(authorization, "Bearer") {
+		t := strings.Split(authorization, " ")
+		if len(t) != 2 {
+			return nil, errInvalidAuthToken.WithAttributes("token", authorization)
+		}
+		token = t[1]
+	}
+	if _, _, _, err := auth.SplitToken(token); err != nil || len(token) != apiKeyLength {
+		return nil, errInvalidAuthToken.WithAttributes("token", authorization)
+	}
```

Comment:
Why do we need to restrict to The Things Stack tokens? What if someone wants to connect their gateway to a different network server?

Reply:
Ok but then we cannot run any validation except the CRLF trimRight.

Reply:
TBH, I don't think this is the place to validate anything. All we need to do here is make sure that there's only a single CRLF in here.